src/main/java/com/google/sps/data/Guide.java

```diff
@@ -49,4 +50,54 @@ public String getCategory() {
     public long getTimestamp() {
         return timestamp;
     }
+
+    public Error getError(){
+        return error;
+    }
+
+    private boolean emptyString(String str){
+        if(str.equals("")) return true;
+        return false;
+    }
+
+    private boolean moreThan1500bytes(String str){
+        if(title.getBytes().length > 1500) return true;
+        return false;
+    }
+
+    public boolean isInvalid(){
+        String errorMsg = "The following parameter cannot be empty: ";
+        String parameter = null;
+
+        if(emptyString(title))
+            parameter = "title";
```

Comment:
I think this would be more readable with a helper method to construct the error.

[return|throw] badParameterError("title", "cannot be empty");

Reply:
Fixed in next commit